cloudflare/resource_cloudflare_access_mutual_tls_certificate.go

```diff
@@ -0,0 +1,195 @@
+			log.Printf("[INFO] Access Mutal TLS Certificate %s no longer exists", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error finding Access Mutual TLS Certificate %q: %s", d.Id(), err)
+	}
+
+	d.Set("name", accessMutualTLSCert.Name)
+	d.Set("associated_hostnames", accessMutualTLSCert.AssociatedHostnames)
+	d.Set("fingerprint", accessMutualTLSCert.Fingerprint)
+
+	return nil
+}
+
+func resourceCloudflareAccessMutualTLSCertificateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	updatedAccessMutualTLSCert := cloudflare.AccessMutualTLSCertificate{
+		ID:   d.Id(),
+		Name: d.Get("name").(string),
+	}
+	updatedAccessMutualTLSCert.AssociatedHostnames = expandInterfaceToStringList(d.Get("associated_hostnames"))
+
+	log.Printf("[DEBUG] Updating Cloudflare Access Mutal TLS Certificate from struct: %+v", updatedAccessMutualTLSCert)
+
+	identifier, err := initIdentifier(d)
+	if err != nil {
+		return err
+	}
+
+	var accessMutualTLSCert cloudflare.AccessMutualTLSCertificate
+	if identifier.Type == AccountType {
+		accessMutualTLSCert, err = client.UpdateAccessMutualTLSCertificate(identifier.Value, d.Id(), updatedAccessMutualTLSCert)
+	} else {
+		accessMutualTLSCert, err = client.UpdateZoneAccessMutualTLSCertificate(identifier.Value, d.Id(), updatedAccessMutualTLSCert)
+	}
+	if err != nil {
+		return fmt.Errorf("error updating Access Mutual TLS Certificate for %s %q: %s", identifier.Type, identifier.Value, err)
+	}
+
+	if accessMutualTLSCert.ID == "" {
+
+	}
+
+	return resourceCloudflareAccessMutualTLSCertificateRead(d, meta)
+}
+
+func resourceCloudflareAccessMutualTLSCertificateDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	certID := d.Id()
+
+	log.Printf("[DEBUG] Deleting Cloudflare Access Mutual TLS Certificate using ID: %s", certID)
+
+	identifier, err := initIdentifier(d)
+	if err != nil {
+		return err
+	}
+
+	if identifier.Type == AccountType {
+		err = client.DeleteAccessMutualTLSCertificate(identifier.Value, certID)
+	} else {
+		err = client.DeleteZoneAccessMutualTLSCertificate(identifier.Value, certID)
+	}
+	if err != nil {
+		return fmt.Errorf("error deleting Access Mutual TLS Certificate for %s %q: %s", identifier.Type, identifier.Value, err)
+	}
+
+	resourceCloudflareAccessMutualTLSCertificateRead(d, meta)
+
+	return nil
+}
+
+func resourceCloudflareAccessMutualTLSCertificateImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	attributes := strings.SplitN(d.Id(), "/", 2)
+
+	if len(attributes) != 2 {
+		return nil, fmt.Errorf("invalid id (\"%s\") specified, should be in format \"accountID/accessMutualTLSCertificateID\"", d.Id())
+	}
+
+	accountID, accessMutualTLSCertificateID := attributes[0], attributes[1]
+
+	log.Printf("[DEBUG] Importing Cloudflare Access Mutual TLS Certificate: id %s for account %s", accessMutualTLSCertificateID, accountID)
+
+	d.Set("account_id", accountID)
```

Comment:
we have the ability to set the zone _and_ account - should we add another identitifer to the import ID to help differentiate which to use? i.e. `account/abc123/cert123` and `zone/def456/cert123`?

Reply:
Yeah, good call. It looks like we need to do this with all of our Access resources -- should we do that in a separate PR?

Reply:
I would fix this resource in this PR but the others are fine to go in another PR. We have #929 to track it as well - I just haven't got it to it just yet :p